VideoLocker/src/main/java/org/edx/mobile/view/Router.java

```diff
@@ -329,4 +331,10 @@ public void showUserProfileEditor(@NonNull Context context, @NonNull String user
     public void showCertificate(@NonNull Context context, @NonNull EnrolledCoursesResponse courseData) {
         context.startActivity(CertificateActivity.newIntent(context, courseData));
     }
+
+    public void showCourseDetail(@NonNull Activity sourceActivity, @NonNull CourseDetail courseDetail) {
+        Intent courseDetailIntent = new Intent(sourceActivity, CourseDetailActivity.class);
```

Comment:
nit: It makes more sense to have logic about how to create a correct intent for a specific activity in that activity itself rather than in the Router (Yes, there are a lot of bad examples in Router already, they should really be cleaned up...)

I usually just put a static method in the activity; see CertificateActivity for example.